src/components/Referrals/AddAffiliateCode.js

```diff
@@ -0,0 +1,170 @@
+            recentlyAddedCodes={recentlyAddedCodes}
+            setRecentlyAddedCodes={setRecentlyAddedCodes}
+          />
+        ) : (
+          <button className="App-cta Exchange-swap-button" type="submit" onClick={connectWallet}>
+            Connect Wallet
+          </button>
+        )}
+      </div>
+    </div>
+  );
+}
+
+export function AffiliateCodeForm({
+  handleCreateReferralCode,
+  recentlyAddedCodes,
+  setRecentlyAddedCodes,
+  callAfterSuccess,
+}) {
+  const [referralCode, setReferralCode] = useState("");
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [error, setError] = useState("");
+  const inputRef = useRef("");
+  const [referralCodeCheckStatus, setReferralCodeCheckStatus] = useState("ok");
+  const debouncedReferralCode = useDebounce(referralCode, 300);
+  const { account, chainId } = useWeb3React();
+
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+
+  useEffect(() => {
+    let cancelled = false;
+    const checkCodeTakenStatus = async () => {
+      if (error) {
+        setReferralCodeCheckStatus("ok");
+        return;
+      }
+      const { status: takenStatus } = await getReferralCodeTakenStatus(account, debouncedReferralCode, chainId);
+      // ignore the result if the referral code to check has changed
+      if (cancelled) {
+        return;
+      }
+      if (takenStatus === "none") {
+        setReferralCodeCheckStatus("ok");
+      } else {
+        setReferralCodeCheckStatus("taken");
+      }
+    };
+    setReferralCodeCheckStatus("checking");
+    checkCodeTakenStatus();
+    return () => {
+      cancelled = true;
+    };
+  }, [account, debouncedReferralCode, error, chainId]);
+
+  function getButtonError() {
+    if (!debouncedReferralCode) {
+      return "Enter a code";
+    }
+    if (referralCodeCheckStatus === "taken") {
+      return "Code already taken";
+    }
+    if (referralCodeCheckStatus === "checking") {
+      return "Checking code...";
+    }
+
+    return false;
+  }
+
+  const buttonError = getButtonError();
+
+  function getPrimaryText() {
+    if (buttonError) {
+      return buttonError;
+    }
+
+    if (isProcessing) {
+      return "Creating...";
+    }
+
+    return "Create";
+  }
+  function isPrimaryEnabled() {
+    if (buttonError) {
+      return false;
+    }
+    if (error || isProcessing) {
+      return false;
+    }
+    return true;
+  }
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    setIsProcessing(true);
+    const { status: takenStatus, info: takenInfo } = await getReferralCodeTakenStatus(account, referralCode, chainId);
+    if (["all", "current", "other"].includes(takenStatus)) {
+      setIsProcessing(false);
+    }
+
+    if (takenStatus === "none" || takenStatus === "other") {
+      const ownerOnOtherNetwork = takenInfo[chainId === ARBITRUM ? "ownerAvax" : "ownerArbitrum"];
+      try {
+        const tx = await handleCreateReferralCode(referralCode);
```

Comment:
sorry missed that last time, can we move `handleCreateReferralCode` to Api/referrals.js?